pkg/util/azure/azure.go

```diff
@@ -43,20 +42,20 @@ func GetHostAlias() (string, error) {
 	return res, nil
 }
 
-// GetClusterName returns the name of the cluster containing the current VM
+// GetClusterName returns the name of the cluster containing the current VM by parsing the resource group name.
+// It expects the resource group name to have the format (MC|mc)_resource-group_cluster-name_zone
 func GetClusterName() (string, error) {
 	all, err := getResponse(metadataURL + "/metadata/instance/compute/resourceGroupName?api-version=2017-08-01&format=text")
 	if err != nil {
 		return "", fmt.Errorf("unable to query metadata endpoint: %s", err)
 	}
 
 	splitAll := strings.Split(all, "_")
-	if len(splitAll) < 4 || splitAll[0] != "MC" {
-		return "", errors.New("cannot parse the clustername from metadata")
+	if len(splitAll) < 4 || strings.ToLower(splitAll[0]) != "mc" {
+		return "", fmt.Errorf("cannot parse the clustername from resource group name: %s", all)
 	}
 
-	clusterName := splitAll[len(splitAll)-2]
-	return clusterName, nil
+	return splitAll[len(splitAll)-2], nil
```

Comment:
nit: that one is a bit scary (the slice read access with a dynamic value which could < 0) but I guess that it has never been an issue as it is from the previous code. If by any chance you have the time to test the slice length that would be awesome, discard if you feel that's not needed.

Reply:
it's protected by the 2 lines above, if `len(splitAll) < 4` (i.e `len(splitAll)-2` could be < 0) we return.
in other words, at line 58 it's guaranteed that `len(splitAll)` >= 4 (i.e `len(splitAll)-2` >= 2)

Reply:
I'm going to merge the PR, I hope my answer above was clear and reassuring